### PR TITLE
AB - Email Notifications - Small Fix

### PR DESCRIPTION
**Pull request by HasnatL-Adapptlimited**
# Description
- DL filtering was changing the dataset filtering before so it has been fixed

## Ticket

- [ ] - DL filtering is no longer affecting the dataset filters

### CODE REVIEW

**Comment by HasnatL-Adapptlimited** [comment]
Hey @AntoineRelief 

PR for small fix has been raised!

Thanks
CC: @morgang-adappt @Vaishali1709

File: libs/shared/src/lib/components/email/components/email-template/email-template.component.ts [hunk 1]
```diff
@@ -1028,7 +1028,7 @@ export class EmailTemplateComponent
   getDataSetPreview(filterType: string) {
     this.noEmail.emit(false);
     this.loading = true;
-    const currentDataset = clone(this.selectedDataset);
+    const currentDataset = JSON.parse(JSON.stringify(this.selectedDataset));
```
Comment (AntoineRelief) [inline libs/shared/src/lib/components/email/components/email-template/email-template.component.ts:1031]: @HasnatL-Adapptlimited 

won't clone work there as well?

Comment (AntoineRelief) [reply libs/shared/src/lib/components/email/components/email-template/email-template.component.ts:1031]: @HasnatL-Adapptlimited

any reason not to use it?
I'm about reviewing the ticket

Comment (HasnatL-Adapptlimited) [reply libs/shared/src/lib/components/email/components/email-template/email-template.component.ts:1031]: We tried using with clone but realised it was overwriting the data, so we needed a deep clone as to not overwrite it

Comment (AntoineRelief) [reply libs/shared/src/lib/components/email/components/email-template/email-template.component.ts:1031]: okay, thanks!

**Comment by AntoineRelief** [comment]
@HasnatL-Adapptlimited 

Okay, I'll merge it but I don't see any issue at all with previous code

it would have been good to have a video or repro steps

**Pull request merged**